src/main/java/com/example/springlabs/repositories/UserRepositoryImpl.java

```diff
@@ -0,0 +1,54 @@
+package com.example.springlabs.repositories;
+
+import com.example.springlabs.model.User;
+import org.springframework.stereotype.Repository;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Repository
+public class UserRepositoryImpl implements UserRepository{
+    private final List<User> users;
+
+    public UserRepositoryImpl(ArrayList<User> users) {
+        this.users = users;
+    }
+
+    @Override
+    public List<User> getUsers() {
+        return users;
+    }
+
+    @Override
+    public void deleteUserById(int id) {
+        users.removeIf(user -> user.getId() == id);
+    }
+
+    @Override
+    public User getUserById(int id) {
+        return users.stream()
+                .filter(user -> user.getId() == id)
+                .findFirst()
+                .orElse(null);
+    }
+
+    @Override
+    public void updateUser(User updatedUser) {
+        for (User user : users) {
+            if (user.getId() == updatedUser.getId()) {
+                user.setName(updatedUser.getName());
+                user.setSurname(updatedUser.getSurname());
+                user.setEmail(updatedUser.getEmail());
+                user.setPassword(updatedUser.getPassword());
+                user.setAdmin(updatedUser.isAdmin());
```

Comment:
I think it is better to insert `break;` here because we need to iterate till the moment we find the one user and then we have no reason to iterate through users.